compiler/modules/front/const.bal

```diff
@@ -79,24 +79,29 @@ function resolveConstDefn(ModuleSymbols mod, s:ConstDefn defn) returns s:Resolve
     }
     else {
         defn.resolved = false;
-        ConstFoldContext cx = new ConstFoldContext(defn, mod);
         s:SubsetBuiltinTypeDesc? td = defn.td;
         t:SemType? expectedType = td == () ? () : resolveBuiltinTypeDesc(td);
-        s:Expr expr = check foldExpr(cx, expectedType, defn.expr);
-        if expr is s:ConstValueExpr {
-            if expectedType == () || t:containsConst(expectedType, expr.value) {
-                s:ResolvedConst r = [t:singleton(expr.value), { value: expr.value }];
-                defn.resolved = r;
-                return r;
-            }
-            else {
-                return err:semantic(`initializer of ${defn.name} is not a subtype of the declared type`, s:defnLocation(defn));
-            }
+        s:ResolvedConst resolvedConst = check resolveConst(mod, defn, defn.expr, expectedType);
+        defn.resolved = resolvedConst;
+        return resolvedConst;
+    }
+}
+
+function resolveConst(ModuleSymbols mod, s:ModuleLevelDefn defn, s:Expr origExpr, t:SemType? expectedType) returns s:ResolvedConst|FoldError {
+    ConstFoldContext cx = new ConstFoldContext(defn, mod);
+    s:Expr expr = check foldExpr(cx, expectedType, origExpr);
+    if expr is s:ConstValueExpr {
+        if expectedType == () || t:containsConst(expectedType, expr.value) {
+            s:ResolvedConst r = [t:singleton(expr.value), { value: expr.value }];
```

Comment:
need spaces around `[`

Reply:
Would something like this work here?
```ballerina
            return <s:ResolvedConst> [ 
                t:singleton(expr.value), 
                { value: expr.value } 
                ];
```

Reply:
I meant
```suggestion
            s:ResolvedConst r = [ t:singleton(expr.value), { value: expr.value } ];
```

Reply:
That's not the style I have used: I put spaces inside curly braces not square or round. Yes, I know this is arbitrary.

Reply:
Ahh yes. please disregard.

Reply:
The line breaks was because it felt like too many spaces.